pkg/k8s/watchers/pod.go

```diff
@@ -276,7 +277,24 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 		// PodIPs update
 		fallthrough
 	case labelsChanged:
-		k.redirectPolicyManager.OnUpdatePod(newK8sPod)
+		lrpNeedsReassign = true
+	}
+
+	oldPodReady := slim_corev1.ConditionUnknown
+	newPodReady := slim_corev1.ConditionUnknown
+
+	for _, cond := range oldK8sPod.Status.Conditions {
+		if cond.Type == slim_corev1.PodReady {
+			oldPodReady = cond.Status
+		}
+	}
+	for _, cond := range newK8sPod.Status.Conditions {
+		if cond.Type == slim_corev1.PodReady {
+			newPodReady = cond.Status
+		}
+	}
```

Comment:
Shouldn't we make the assignment of the `*PodReady` variables conditional? The way the code is written right now it will make the pod to be considered "Ready" even if the first condition was not "Ready" and the last "Condition" was "Ready".

Reply:
I'm a bit confused, the statues variables (`oldPodReady` and `newPodReady`) are already conditioned on the *type* of the condition we are checking, not sure why other conditions would set them?

Reply:
@Weil0ng oh, I think I missed the `cond.Type == slim_corev1.PodReady`, so we are expecting that there's a single condition of type PodReady? If yes, couldn't we have a "break" on lines 288 and 294?

Reply:
yes, done.